STRINGExplore: Interactive exploration of STRING interactions and associated protein activity

gabmorg opened the pull request:
This tool obtains an interactive network from the input of any gene list, through querying of a (global) high confidence STRING combined score edge list. It uses these edges to find and relate protein action information (also obtained by mapped STRING data) to discovered high-confidence edges in the input system (expected to be a curated gene system and so likely to have high-confidence edges associated with it). Betweenness centrality and tooltips are used to create the network and group the nodes in interactive subsystems.

Future: I'd like to create an option to use coexpression scores from Dr. Steipe's GEO analysis to make the betweenness centrality model more meaningful biologically. I'd also like to create more flexibility in the code for the data sources to allow future iterations of STRING (or even third-party sources) to be compatible with this tool

gabmorg committed:
Prepared datasets, initial code for example

gabmorg committed:
updated vignette

gabmorg committed:
mapping the proteins actions data

gabmorg committed:
includes all code required for package function, in script form

gabmorg committed:
Add authorship, imports, and ORCID ID data to DESCRIPTION and vignette

gabmorg committed:
functionalized example of PHALY with STRINGExplore ESA tool instead of script!

gabmorg committed:
Added comments to mapping functions needed for preparing the data for the example results

gabmorg committed:
commented functionalized package tool

gabmorg committed:
fixed typo in ORCID field of DESCRIPTION

gabmorg committed:
removed igraph due to warning's it's already been incorporated into R, and fixed some var naming in tool functions

gabmorg committed:
add bioMart to req

gabmorg committed:
remove unused packages, add loading of data to tool file

gabmorg committed:
removed default data for actions and edges in ESA definition file

gabmorg committed:
commented out example results code to test if Travis just hates that I pushed RData files

gabmorg committed:
comment out the mapping code because I'm pretty sure it shouldn't be included anyway, and it's failing beacuse I'm not pushing the original data files

gabmorg committed:
adding BiocCheck as dependency

gabmorg committed:
successful build locally, removed RData files from /data

gabmorg committed:
committing to merge changes from updated functions

gabmorg committed:
conflicts w updates to upstream

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	NAMESPACE

gabmorg committed:
Package can be built and installed locally, and uses fetchComponents and fetchData rather than local RDS files. All unnecessary code for STRINGExplore.R moved to /dev and not pushed

gabmorg committed:
/dev directory to save code for mapping and RDS/RData files just in case. Not needed for package build. Deleted large data files

gabmorg committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	NAMESPACE

gabmorg committed:
working on vignette: initial notes

gabmorg committed:
working on vignette: initial notes

gabmorg committed:
forgot to save all my vignette changes

Merge branch 'master' of https://github.com/gabmorg/BCB420.2019.ESA

# Conflicts:
#	vignettes/BCB420.2019.STRINGExplore.Rmd

sapirl commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,136 @@
+  geneSysEdges <- STRINGedges[sel, c("protein1", "protein2")]
+
+  #### 2. igraph setup (visNetwork backbone)####
+  sXG <-  graph.data.frame(geneSysEdges, directed = F)
```
Referring to the package this function belongs to ("igraph::")

gabmorg replied:
used the @import tag approach for the ROxygen header instead

sapirl commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,136 @@
+
+  #### 2. igraph setup (visNetwork backbone)####
+  sXG <-  graph.data.frame(geneSysEdges, directed = F)
+  sXGgraph <-  simplify(sXG)
```
Referring to the package this function belongs to ("igraph::")

sapirl commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,136 @@
+  sXGgraph <-  simplify(sXG)
+
+  #### 3. Calculate betweeness centrality of graph based on STRING scores ####
+  bC <-  centr_betw(sXG)
```
Referring to the package this function belongs to

gabmorg replied:
used the @import tag approach for the ROxygen header instead

sapirl commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,136 @@
+  bC <-  centr_betw(sXG)
+  nodeBetw <- bC$res
+  nodeBetw <- round(log(nodeBetw + 1)) + 1
+  V(sXGgraph)$btwndegree <- nodeBetw
```
Referring to the package this function belongs to

sapirl commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,136 @@
+
+  #### 4. visNetwork setup from igraph object ####
+  # 4.1 Nodes dataframe setup
+  nodes <- get.data.frame(sXGgraph, what="vertices")
```
Referring to the package this function belongs to

sapirl commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,136 @@
+  nodes <- nodes[order(nodes$id, decreasing = F),]
+
+  # 4.2 Edges dataframe setup
+  edges <- get.data.frame(sXGgraph, what="edges")
```
Referring to the package this function belongs to

sapirl commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,136 @@
+  edges$title = paste("<p>",edges$from, "--",edges$to, ":", edges$interactions, "</p>")
+
+  #### 5. Plot visNetwork ####
+  network <- visNetwork(nodes, edges, height = "500px", width = "100%") %>%
```
Referring to the package this function belongs to

sapirl commented on dev/mapSTRING.R:
```diff
@@ -0,0 +1,151 @@
+# Script to load STRING coexpression data and map it
```
I can't recall whether Dr. Steipe said that he will upload it or not,
If it is uploaded by him through a different file, don't forget to remove yours!

hyginn replied:
This is a part of the documentation. Two options: move this to ./inst/scripts after merge, or cover it in a vignette. ./dev/ is only for actual development.

gabmorg committed:
commit conflict resolution to allow merge of new SysDB fns

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	NAMESPACE

gabmorg committed:
Merge remote-tracking branch 'upstream/master'

importing updates to SysDB.. fns.

gabmorg committed:
example result and addressing change requests to allow merge

gabmorg committed:
Merge remote-tracking branch 'upstream/master'

updates from remote

gabmorg committed:
remove test code

gabmorg reviewed (commented):
Added package::function notation as requested by reviewer, tested STRINGExplore with new code, removed testing code from within fn

gabmorg commented:
@sapirl addressed changes to code and Travis checks pass. Currently working on vignette and tests, so core code will not be touched. Are you able to merge yet or is that only after the documentations steps?

gabmorg committed:
add stabilization for the network, so it doesn't "buzz" when it takes a little longer to come up with the appropriate layout. Completed STRINGactions vignette

gabmorg committed:
STRINGedges0.8 vignette

gabmorg committed:
Merge branch 'master' into master

gabmorg committed:
pull changes from merged code on repo, push updates to vignettes

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	DESCRIPTION
#	NAMESPACE

gabmorg committed:
merge conflicts

Merge branch 'master' of https://github.com/gabmorg/BCB420.2019.ESA

# Conflicts:
#	DESCRIPTION
#	NAMESPACE

gabmorg committed:
finished STRINGExplore vignette

gabmorg committed:
add needed commas to DESCRIPTION

gabmorg committed:
DESCRIPTION file added? Travis not building. Committing to see if this would change anything on remote test

gabmorg reviewed (commented):
This is able to automatically merge. Do you mind do that so that I can pull a functional build setup to push my new vignette to?

gabmorg committed:
update STRINGEXplore vignette and delete image. This is a rollback because the build was broken in subsequent commits

gabmorg committed:
Merge branch 'master' of https://github.com/gabmorg/BCB420.2019.ESA

gabmorg committed:
package build with my code and the tool works!

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	DESCRIPTION
#	NAMESPACE

gabmorg committed:
committing more conflicts from upstream. Build and works!

Merge branch 'master' of https://github.com/gabmorg/BCB420.2019.ESA

# Conflicts:
#	DESCRIPTION
#	NAMESPACE

gabmorg committed:
Merge branch 'master' into master

gabmorg committed:
Merge branch 'master' into master

hyginn commented on BCB420.2019.ESA.Rproj:
```diff
@@ -18,3 +18,4 @@ StripTrailingWhitespace: Yes
 BuildType: Package
```
`.Rproj` should not be here. Removed.

hyginn committed:
Delete BCB420.2019.ESA.Rproj

hyginn committed:
Delete DESCRIPTION

Probably not safe to merge at this stage. Removed, will recreate changes manually.

hyginn committed:
Delete NAMESPACE

Should be recreated by hand on local build.

hyginn commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,144 @@
+  #### 1. Prepare / Load data:####
+  # alternate / future route: add option to create edges from
+  # high correlation scored expression data (Pearson rho >80%) and network those
+  STRINGactions <- fetchData("STRINGactions")
```
Data load should not be done inside functions, and especially not with hardcoded names. Needs to load externally and pass the objects. The closest acceptable approximation to the code as used here would be:
```R
STRINGExplore <- function(systemName, STRINGedges = fetchData("STRINGedges0.8") ) { ...
```

hyginn commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,144 @@
+  STRINGedges <- fetchData("STRINGedges0.8")
+  geneSys <- list()
+
+  dat <- SyDBgetSysSymbols(fetchData("SysDB"), systemName)
```
Same here: needs to be passed from parameter list

hyginn commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,144 @@
+  sXG <-  igraph::graph.data.frame(geneSysEdges, directed = F)
+  sXGgraph <-  igraph::simplify(sXG)
+
+  #### 3. Calculate betweeness centrality of graph based on STRING scores ####
```
* betweenness

hyginn commented on R/STRINGExplore.R:
```diff
@@ -0,0 +1,144 @@
+  geneSys <- unlist(dat)
+  names(geneSys) <- c()
+
+  actionSet <- STRINGactions[STRINGactions$protein1 %in% geneSys &
```
better: use a `sel <- ...` idiom here too.

hyginn commented on dev/STRINGExplore_PHALY-Example.R:
```diff
@@ -0,0 +1,31 @@
+#### Versions ####
```
Two options: move this to `./inst/scripts` after merge, or cover it in the vignette. `./dev/` is only for actual development.

hyginn commented on dev/mapSTRING.R:
```diff
@@ -0,0 +1,151 @@
+validationActions <- apply(STRINGactions[ , c("protein1", "protein2")], 1, sPaste)
+validate <- intersect(validationActions, validationEdges)
+View(validate)
+length(validate) # 6161 edges with logged protein actions
```
coding style: `# [END]`

hyginn reviewed (commented):
This is a significant effort, and I am looking forward to see it in action. However I need to cherry pick the files. To avoid issues with other things having gone out of sync, I'll do that manually.

hyginn commented:
Well - no surprise. Once removing the files that would cause issues after merging, the checks fail. Closing this and merging manually.
